fire/FatesFuelClassesMod.F90

```diff
@@ -0,0 +1,63 @@
+module FatesFuelClassesMod
+
+  use FatesLitterMod, only : ncwd
+  
+  implicit none
+  private
+
+  integer, parameter, public :: nfsc = 6 ! number of total fuel classes  
+    
+  type :: fuel_classes_type
+    ! There are six fuel classes:
+    ! 1) twigs, 2) small branches, 3) large branches 4) trunks
+    ! 5) dead leaves, 6) live grass
+    integer, private :: twigs_i = 1          ! array index for twigs pool
+    integer, private :: small_branches_i = 2 ! array index for small branches pool
+    integer, private :: large_branches_i = 3 ! array index for large branches pool
+    integer, private :: dead_leaves_i = 5    ! array index for dead leaves pool
+    integer, private :: live_grass_i = 6     ! array index for live grass pool
+    integer, private :: trunks_i = 4         ! array index for trunks pool
+
+    contains 
+
+    procedure :: twigs, small_branches, large_branches, trunks
+    procedure :: dead_leaves, live_grass
+
+  end type fuel_classes_type
+
+  ! actual type we can pass around 
+  type(fuel_classes_type), public :: fuel_classes
+
+  contains
+
+  integer function twigs(this)
+    class(fuel_classes_type), intent(in) :: this 
+    twigs = this%twigs_i
+  end function twigs 
+
+  integer function small_branches(this)
+    class(fuel_classes_type), intent(in) :: this 
+    small_branches = this%small_branches_i
+  end function small_branches 
+
+  integer function large_branches(this)
+    class(fuel_classes_type), intent(in) :: this 
+    large_branches = this%large_branches_i
+  end function large_branches 
+
+  integer function trunks(this)
+    class(fuel_classes_type), intent(in) :: this 
+    trunks = this%trunks_i
+  end function trunks 
+
+  integer function dead_leaves(this)
+    class(fuel_classes_type), intent(in) :: this 
+    dead_leaves = this%dead_leaves_i
```

Comment:
it's just a getter function. This is a (Fortran-limited) way of carrying the fuel class indices around in a class (rather than having to read each of them in), but still protecting their values.

You unfortunately can't have a protected class variable, only private - which means you have to use a method to get the value.

The setter/getter pattern is fairly common in object-oriented programming. It's a little clunky in Fortran, unfortunately. I was trying it out here but can go back to what we had before if people don't like it. I just found it a lot nicer to just `use` the fuel_classes class rather than each parameter

Reply:
The other option would be to just have a module with just the fuel classes as constants/parameters in it and allow the use of the whole module, which is effectively the same idea, but goes against our convention of having the use, : only, and obfuscates the origin of each of the constants.

Reply:
I am continually frustrated that Fortran doesn't have an `enumerator` type....

Reply:
I think since they are constants, they can be defined as "integer, parameter :: ", which makes them immutable.  I have no problem with getters/setters though. It adds one small extra step when someone who is reading the code is tracking what the value, but that's trivial.

Reply:
whoa, I can't believe I never ran into that. weird